tests/secretCRUD_test.go

```diff
@@ -54,22 +55,32 @@ func Test_SecretCRUD(t *testing.T) {
 		t.Errorf("got %v, wanted %s in slice", secrets, setName)
 	}
 
-	// Docker Swarm secrets are immutable, so skip the update tests for swarm.
-	if config.SecretUpdate {
-		newValue := "this-is-the-edited-secret-value"
-		updateStatus, _ := config.Client.UpdateSecret(ctx, types.Secret{Name: setName, Value: newValue})
-		if updateStatus != http.StatusOK && updateStatus != http.StatusAccepted {
-			t.Errorf("got %d, wanted %d or %d", updateStatus, http.StatusOK, http.StatusAccepted)
-		}
-		t.Logf("Got correct response for updating secret: %d", updateStatus)
+	newValue := "this-is-the-edited-secret-value"
+	updateStatus, _ := config.Client.UpdateSecret(ctx, types.Secret{Name: setName, Value: newValue})
+	if updateStatus != http.StatusOK && updateStatus != http.StatusAccepted {
+		t.Errorf("got %d, wanted %d or %d", updateStatus, http.StatusOK, http.StatusAccepted)
+	}
+	t.Logf("Got correct response for updating secret: %d", updateStatus)
 
-		// Verify that the secret value was edited.
-		value = string(invoke(t, functionRequest, "", "", http.StatusOK))
-		if value != setValue {
-			t.Errorf("got %s, wanted %s", value, newValue)
+	if config.ProviderName == faasNetesProviderName {
```

Comment:
Since, secret update does not happen directly we needed to scale down and up the function for secret update